### PR TITLE
File viewer - add generic read-only dialog

### PR DESCRIPTION
**Pull request by crayment**
### ✨ Features
- Added a generic read-only file viewer that can open any absolute text file path with Birdhouse theming and syntax highlighting.
- Markdown files now open in rich mode by default with a `Rich | Raw` toggle, while non-markdown files open directly in raw/code view.
- Supporting files in the Skills Library are now clickable and open in the generic file viewer.
- Added a reveal-in-Finder action to the file viewer so the currently viewed file can be opened in the OS file manager.

### 🐛 Fixes
- Fixed Skills Library list scrolling so auto-scroll only happens for programmatic opens from skill references, not for normal user list clicks.
- Fixed the skill detail pane so switching to a different skill resets the right-side scroll position back to the top.
- Fixed modal layering so the file viewer presents above the Skills Library instead of appearing behind it.

---

### 🏗️ Infrastructure
- Extended the existing workspace `/files` route family with read-only file viewing and reveal endpoints.
- Added server-side file viewing guards for absolute paths, regular files, text-only reads, and size-limited content loading.

### 🧹 Code Quality
- Introduced a small `frontend/src/file-viewer/` feature folder for the reusable dialog, API client, path helper, and tests.
- Kept the file viewer decoupled from Skills-specific state so it can be reused by other parts of the app later.
- Refined Skills Library selection behavior with explicit programmatic scroll intent instead of broad reactive auto-scroll triggers.

## Testing
- `bun test src/routes/files.test.ts`
- `bun run typecheck` in `projects/birdhouse/server`
- `bun run lint` in `projects/birdhouse/server`
- `bun --bun vitest run src/file-viewer/components/FileViewerDialog.test.tsx src/skills/components/SkillDetailContent.test.tsx src/skills/components/SkillDetailModal.test.tsx src/skills/components/SkillLibraryDialog.test.tsx src/skills/components/SkillListPane.test.tsx`
- `bun run typecheck` in `projects/birdhouse/frontend`
- `bun run lint` in `projects/birdhouse/frontend`
- Manual isolated Birdhouse browser testing with screenshots and MP4 capture for:
  - generic file viewer happy path
  - modal layering retest
  - markdown rich/raw toggle
  - non-markdown raw/code view

### CODE REVIEW

File: projects/birdhouse/server/src/routes/files.ts [hunk 1]
```diff
@@ -1,12 +1,52 @@
...
+
+export function createFileRoutes(options: CreateFileRoutesOptions = {}) {
   const app = new Hono();
+  const revealFile = options.revealFileInFileManager ?? revealFileInFileManager;
+
+  app.get("/view", async (c) => {
+    const filePath = c.req.query("path");
+
+    try {
+      return c.json(readViewableFile(filePath ?? ""));
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/server/src/routes/files.ts:21]: @cubic-dev-ai We reviewed this one and are treating the arbitrary absolute-path read in `/files/view` as an intentional product tradeoff for Birdhouses local app model, so we are not adding a workspace-only or allowlist restriction here.

The viewer is meant to stay generic and support opening skill-related files that can live outside the current workspace root, including global skill locations. We agree this would be a different call in a remotely exposed or multi-tenant service, but for this local desktop/server setup were keeping the broader path behavior intentionally.

File: projects/birdhouse/server/src/routes/files.ts [hunk 2]
```diff
@@ -1,12 +1,52 @@
...
+        return c.json({ error: error.message }, statusByCode[error.code]);
+      }
+
+      throw error;
+    }
+  });
+
+  app.post("/reveal", async (c) => {
+    const body = await c.req.json();
+    const filePath = typeof body.path === "string" ? body.path : "";
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/server/src/routes/files.ts]: @cubic-dev-ai Fixed in `616bc6c0fbb93ea2c63448e6ad24b6e69b6f120c`.

`/files/reveal` now catches malformed JSON and returns `400 { error: "Invalid JSON body" }`, and it no longer throws on `null` or other non-object payloads before validating `path`. I also added focused route tests covering both malformed JSON and `null` request bodies.

File: projects/birdhouse/frontend/src/skills/components/SkillDetailContent.tsx [hunk 3]
```diff
@@ -146,122 +155,173 @@ const SkillDetailContent: Component<SkillDetailContentProps> = (props) => {
...
+      setViewedFile(file);
+    } catch (err) {
+      setViewedFile(null);
+      setViewerError(err instanceof Error ? err.message : "Failed to load file");
+    } finally {
+      setViewerLoading(false);
+    }
+  };
+
+  createEffect(() => {
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/frontend/src/skills/components/SkillDetailContent.tsx:194]: @cubic-dev-ai Fixed in `6b6fbe0d93e7381d69a578db41d8dde93d483c94`.

I now clear the nested file-viewer state when `SkillDetailContent` switches to a different skill, so an open supporting file cannot leak across skill changes.

File: projects/birdhouse/frontend/src/skills/components/SkillLibraryDialog.tsx [hunk 4]
```diff
@@ -137,11 +153,12 @@ const SkillLibraryDialog: Component<SkillLibraryDialogProps> = (props) => {
     });
   });
 
-  const selectSkill = (skillId: string | null) => {
+  const selectSkill = (skillId: string | null, options?: { autoScroll?: boolean }) => {
     const nextId = skillId || "main";
     if (skillId) {
       setStoredSelectedSkillId(skillId);
     }
+    setAutoScrollSkillId(options?.autoScroll && skillId ? skillId : null);
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/frontend/src/skills/components/SkillLibraryDialog.tsx:161]: @cubic-dev-ai Fixed in `6b6fbe0d93e7381d69a578db41d8dde93d483c94`.

I added request invalidation around the queued auto-scroll work and clear pending programmatic auto-scroll state on later search and scope-filter changes, so stale requests cannot fire after the UI has moved on.

File: projects/birdhouse/frontend/src/skills/components/SkillDetailContent.tsx [hunk 5]
```diff
@@ -146,122 +155,173 @@ const SkillDetailContent: Component<SkillDetailContentProps> = (props) => {
...
   };
 
+  const handleOpenSupportingFile = async (relativeFilePath: string) => {
+    const absoluteFilePath = resolveSiblingFilePath(props.skill.location, relativeFilePath);
+    setViewerOpen(true);
+    setViewerLoading(true);
+    setViewerError(null);
+
+    try {
+      const file = await fetchFileViewerContent(props.workspaceId, absoluteFilePath);
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/frontend/src/skills/components/SkillDetailContent.tsx:174]: @cubic-dev-ai Fixed in `6b6fbe0d93e7381d69a578db41d8dde93d483c94`.

Supporting-file loads are now guarded by a latest-request token, so older async responses cannot overwrite a newer selection.

File: projects/birdhouse/frontend/src/file-viewer/components/FileViewerDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  });
+
+  const handleReveal = async () => {
+    if (!props.workspaceId || !props.file?.path || isRevealing()) {
+      return;
+    }
+
+    setIsRevealing(true);
+    try {
+      await revealFileViewerPath(props.workspaceId, props.file.path);
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/frontend/src/file-viewer/components/FileViewerDialog.tsx:53]: @cubic-dev-ai Fixed in `6b6fbe0d93e7381d69a578db41d8dde93d483c94`.

Reveal-in-Finder failures are now caught in the dialog and surfaced through local error UI instead of rejecting unhandled from the async click path.

File: projects/birdhouse/frontend/src/file-viewer/components/FileViewerDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  error?: string | null;
+}
+
+const FileViewerDialog: Component<FileViewerDialogProps> = (props) => {
+  const baseZIndex = useZIndex();
+  const resolvedTheme = createMemo(() => resolveCodeTheme(codeTheme(), isDark()));
+  const [markdownMode, setMarkdownMode] = createSignal<MarkdownViewMode>("rich");
+  const [isRevealing, setIsRevealing] = createSignal(false);
+
+  createEffect(() => {
```
[the remark this answers is not in the document]
Comment (crayment) [reply projects/birdhouse/frontend/src/file-viewer/components/FileViewerDialog.tsx:33]: @cubic-dev-ai Fixed in `6b6fbe0d93e7381d69a578db41d8dde93d483c94`.

The file viewer now resets markdown mode on dialog open, so reopening the same markdown file starts in `Rich` again.
